### PR TITLE
Add OAuth docs

### PR DESCRIPTION
**Pull request by adrianmroz-allegro**

### CODE REVIEW

File: docs/oauth.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+# OAuth integration
+
+* TOC
+  {:toc}
+  
+## Overview
+
+Turnilo can integrate with your OAuth provider. 
+Wurnilo will redirect your users to authentication page, handle token exchange and will decorate all requests with access token.
```
Comment (mkuthan) [inline docs/oauth.md:9]: s/W/T/

File: docs/oauth.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
+# OAuth integration
+
+* TOC
+  {:toc}
+  
+## Overview
+
+Turnilo can integrate with your OAuth provider.
```
Comment (mkuthan) [inline docs/oauth.md:8]: Could you add information about supported integration method: PKCE?

File: docs/oauth.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+## Configuration
+
+To enable OAuth support you need to add top level `oauth` option in your config with following fields:
+
+- `clientId`: OAuth Client Identifier for your Turnilo instance
+- `tokenEndpoint`: Full address of your [token endpoint](https://tools.ietf.org/html/rfc6749#section-3.2)
+- `authorizationEndpoint`: Full address of your [authorization endpoint](https://tools.ietf.org/html/rfc6749#section-3.1)
+- `redirectUri`: Full address of your Turnilo installation. If you go to this address in browser, your Turnilo install should launch and show home view.
+- `tokenHeaderName`: Name of your choosing for the OAuth token header. You will be reading this header in your server plugin.
```
Comment (mkuthan) [inline docs/oauth.md:19]: Do we have default for header name?

**Review by mkuthan — CHANGES_REQUESTED** [review]

File: docs/oauth.md (hunk 3 above)
Comment (adrianmroz-allegro) [reply docs/oauth.md:19]: No and I don't think we should. This name is used in admin's plugin and should be chosen deliberately. 

I know that you like to have defaults for most of the config options but I think there's no good one here and we need to force admin to pick one.

**Review by mkuthan — APPROVED** [review]

**Pull request merged**